### DISCUSSION TITLE
Building call graph with SootUp from jar file generated by Maven package

### DISCUSSION THREAD

**Discussion by Diwu-Yi** [soot-oss/SootUp#568] [Q&A] [answered]
Hi everyone,

I am new to SootUp and is attempting to build a call graph for a repository I hope to understand more. The repository is[ jsoup](https://github.com/jhy/jsoup) . After generating a snapshot of a version of jar file, I supplied the location to SootUp, and completed the process of call graph construction with something like this:
```
AnalysisInputLocation<JavaSootClass> inputLocation =
                new JavaClassPathAnalysisInputLocation(
                        "/Users/my-name/jsoup-1.11.3-SNAPSHOT.jar");

        JavaLanguage language = new JavaLanguage(8);

        JavaProject project =
                JavaProject.builder(language)
                        .addInputLocation(inputLocation)
                        .addInputLocation(
                                new JavaClassPathAnalysisInputLocation(
                                        "/Library/Java/JavaVirtualMachines/jdk1.8.0_333.jdk/Contents/Home/jre/lib/rt.jar"))
                        .build();

        JavaView view = project.createFullView();

        ViewTypeHierarchy typeHierarchy = new ViewTypeHierarchy(view);
       

        ClassType classDeclarationType = project.getIdentifierFactory().getClassType("org.jsoup.Jsoup");
        ClassType returnType = project.getIdentifierFactory().getClassType("org.jsoup.nodes.Document");
        ClassType type1 = project.getIdentifierFactory().getClassType("java.nio.file.Files");
        ClassType type2 = project.getIdentifierFactory().getClassType("java.lang.String");
        ClassType type3 = project.getIdentifierFactory().getClassType("java.lang.String");
        ArrayList<ClassType> params = new ArrayList<>(3);
        params.add(type1);
        params.add(type2);
        params.add(type3);
        MethodSignature entryMethodSignature =
                JavaIdentifierFactory.getInstance()
                        .getMethodSignature(
                                classDeclarationType,
                                JavaIdentifierFactory.getInstance()
                                        .getMethodSubSignature(
                                                "parse", returnType, params));
        CallGraphAlgorithm cha = new ClassHierarchyAnalysisAlgorithm(view, typeHierarchy);

        CallGraph cg = cha.initialize(Collections.singletonList(entryMethodSignature));
```
And the parameters supplied to construct MethodSignature are copied and pasted from jhy/jsoup 's class headers. I attach a screenshot of the class containing the parse method for your easier reference here.
<img width="947" alt="Screenshot 2023-03-30 at 1 12 50 AM" src="https://user-images.githubusercontent.com/59159857/228616299-65317bb4-6575-465d-af54-bf219e185385.png">

However, when I print out the cg with: cg.callsFrom(entryMethodSignature).forEach(System.out::println);
the result is empty.

Please may I ask does this indeed imply that the parse method is not called by any other method in jsoup or did I do something wrongly ?  
Would really appreciate a confirmation or correction because I am unsure if the MethodSignature is used correctly here and this is my first try. 

Thank you in advance for your time and help.

**Comment by JonasKlauke** [comment, marked as answer]
Hi Diwu-Yu
your parameterlist in the method signature is wrong. the parse method has two String parameters, but your params list has 3 entries. It should work if you remove type1.

**Reply by Diwu-Yi** [reply]
HI JonasKlauke,

Thanks for the reply, it helps.
